javaslang/src/test/java/javaslang/MatchTest.java

```diff
@@ -219,6 +219,26 @@ public void shouldDecomposeSomeSomeTuple() {
 
     // -- List
 
+    @Test
+    public void shouldDecomposeEmptyList() {
```

Comment:
Decomposition, more specifically pattern matching, does only work for algebraic data types. Wikipedia has a good explanation (https://en.wikipedia.org/wiki/Algebraic_data_type), pattern matching is mentioned in the 4th paragraph.

An empty PriorityQueue has no head. Therefore we can't decompose it into a `Tuple2(head, tail)`.